Source/WebCore/editing/FrameSelection.cpp

```diff
@@ -1775,6 +1776,14 @@ void FrameSelection::willBeRemovedFromFrame()
         CursorAlignOnScroll::IfNeeded, TextGranularity::CharacterGranularity);
     m_previousCaretNode = nullptr;
     m_typingStyle = nullptr;
+
+    RefPtr document = protectedDocument();
+    if (RefPtr page = document->page()) {
+        page->focusController().resetFocusedElement();
+#if ENABLE(CONTEXT_MENUS)
+        page->contextMenuController().destroyContext();
+#endif
+    }
 }
```

Comment:
This code is added just to make writing a document leak test using iframes possible for this kind of leak. The focused element keeps a strong ref to a node if a context click occurs within an input element, so we need to clear it.
We also destroy the context here because EventSendingController::contextClick() does not close the context menu unlike a real user would, so the context stays alive. Other tests also assume that the menu is open to test actions like clicking on the menu. 
It's not ideal that we are adding code here just to make this test work right, so I am open to suggestions on how to fix this. We can destroy the context in `Internals::resetToConsistentState(Page& page)`, which makes --world-leaks tool not falsely fail a test, but that doesn't fix an iframe-based test because resetToConsistentState is only called once after the test case has finished running and not for every iframe.

Reply:
I think I should modify EventSendingController::contextClick() to make it not possible to call contextClick() more than once without calling into `ContextMenuController::didDismissContextMenu()` where we destroy the context. As a user, it's not possible to open the context menu again without dismissing the context menu first (meaning the previous context menu the user opened goes away and a new one opens). Therefore, we should make Internals EventSendingController::contextClick() do the same. This would also address the problem above.

Reply:
My preference would be to modify` spelling/disable-automatic-spelling-correction-context-menu-item.html` to do another click somewhere to close the context menu after opening it and also modify Internals::resetToConsistentState() to destroy the context. 
That way, we do not get falsely flagged document leaks for tests that call JS contextClick() in the future and we can also do an accurate iframe-based document leak test, but just for this specific test case.